### PR TITLE
Support transforms for display list optimization. Prevents clipping in #6643.

### PR DESCRIPTION
**Pull request by glennw**
[Review on Reviewable](https://reviewable.io/reviews/servo/servo/6983)

### CODE REVIEW

**Comment by pcwalton** [comment]
Why is DL optimization needed to clip properly?

**Comment by glennw** [comment]
When I say clipping, I'm referring to items being culled incorrectly from some tiles without this change, which gives the appearance that an item is being clipped at tile boundaries. If that makes it clearer?

**Comment by pcwalton** [comment]
Ah, I see. So we were not taking transforms into account but were attempting to optimize anyway.

**Comment by glennw** [comment]
Yep, that's right.

File: components/gfx/display_list/mod.rs [hunk 1]
```diff
@@ -309,14 +302,16 @@ impl StackingContext {
...
                 display_list.print_items("*".to_owned());
             }
 
             // Sort positioned children according to z-index.
             let mut positioned_children: SmallVec<[Arc<StackingContext>; 8]> = SmallVec::new();
             for kid in display_list.children.iter() {
-                positioned_children.push((*kid).clone());
+                if kid.layer.is_none() {
+                    positioned_children.push((*kid).clone());
+                }
```
Comment (pcwalton) [inline components/gfx/display_list/mod.rs]: Why this change?

Comment (glennw) [inline components/gfx/display_list/mod.rs]: It's a slight optimization - the if checks below previously excluded them anyway, this just avoids adding them to the smallvec at all.
